ts/features/bonus/bpd/components/bpdCardComponent/BpdCardComponent.tsx

```diff
@@ -127,29 +128,94 @@ const styles = StyleSheet.create({
     borderBottomWidth: 15,
     borderBottomColor: "rgba(0,0,0,0.1)",
     width: "100%"
+  },
+  fireworksIcon: {
+    width: 16,
+    height: 16,
+    resizeMode: "cover"
   }
 });
 
 type BadgeDefinition = {
   label: string;
 };
 
+type IconType = "closedLock" | "openLock" | "fireworks" | undefined;
+
 type GraphicalState = {
   amount: ReadonlyArray<string>;
   isInGracePeriod: boolean;
-  showLock: boolean;
   statusBadge: BadgeDefinition;
 };
 
 const initialGraphicalState: GraphicalState = {
   amount: ["0", "00"],
   isInGracePeriod: false,
-  showLock: false,
   statusBadge: {
     label: "-"
   }
 };
 
+const iconMap = new Map<IconType, React.ReactNode>([
+  [
+    "closedLock",
+    <IconFont
+      name="io-locker-closed"
+      size={16}
+      color={IOColors.white}
+      key={"closedLocker"}
+    />
+  ],
+  [
+    "openLock",
+    <IconFont
+      name="io-locker-open"
+      size={16}
+      color={IOColors.white}
+      key={"openLocker"}
+    />
+  ],
+  [
+    "fireworks",
+    <Image
```

Comment:
Since PR #2630 we added this as an icon `io-fireworks`, maybe we may uniform the code using the icon instead of image i guess 😉